### PR TITLE
support mTLS, passing Cert and Key to http client

### PR DESCRIPTION
**Pull request by hao1939**
fortio has the options '-cert' and '-key', but it's not used by HTTP client.
The commit passing the options to HTTP client and setup the client side certificate.

### CODE REVIEW

File: fhttp/http_client.go [hunk 1]
```diff
@@ -388,9 +391,20 @@ func NewStdClient(o *HTTPOptions) *Client {
...
+	if o.https {
+		tr.TLSClientConfig = &tls.Config{}
+		if o.Insecure {
+			log.LogVf("using insecure https")
+			tr.TLSClientConfig.InsecureSkipVerify = true
+		}
+		if len(o.Cert) > 0 && len(o.Key) > 0 {
+			cert, err := tls.LoadX509KeyPair(o.Cert, o.Key)
+			if err != nil {
+				log.LogVf(fmt.Sprintf("%v", err))
```
Comment (ldemailly) [inline fhttp/http_client.go]: this probably should be a log.Err (maybe even fatal)

**Comment by ldemailly** [comment]
ping

File: fhttp/http_client.go (hunk 1 above)
Comment (hao1939) [reply fhttp/http_client.go]: got you, changed to log.Err.

**Comment by ldemailly** [comment]
one issue is this isn't plugged into/exercisable from the command line

**Comment by ldemailly** [comment]
actually I was wrong it's fine for client side, will merge (would be nice to have client side too but I will get to that one)

thanks for the contribution! (sorry for the delay, though ideally a test would have been great too)

**Review by ldemailly — APPROVED** [review]

File: fhttp/http_client.go [hunk 2]
```diff
@@ -167,6 +167,9 @@ type HTTPOptions struct {
 	FollowRedirects   bool // For the Std Client only: follow redirects.
 	initDone          bool
 	https             bool   // whether URLSchemeCheck determined this was an https:// call or not
+	Cacert            string // `Path` to a custom CA certificate file to be used
```
Comment (ldemailly) [inline fhttp/http_client.go:170]: this isn't used by the code below :(

**Pull request merged**